adele-payment-manager/src/main/java/com/microservicesteam/adele/payment/paypal/PaypalProxy.java

```diff
@@ -0,0 +1,36 @@
+package com.microservicesteam.adele.payment.paypal;
+
+import org.springframework.stereotype.Component;
+
+import com.paypal.api.payments.Payment;
+import com.paypal.api.payments.PaymentExecution;
+import com.paypal.base.rest.APIContext;
+import com.paypal.base.rest.PayPalRESTException;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+@Component
+public class PaypalProxy {
+
+    private final PaypalConfig.PaypalProperties paypalProperties;
+
+    public Payment create(Payment paymentRequest) throws PayPalRESTException {
+
+        APIContext apiContext = getApiContext();
+        return paymentRequest.create(apiContext);
+    }
+
+    public Payment execute(String paymentId, PaymentExecution paymentExecution) throws PayPalRESTException {
+        Payment payment = new Payment();
```

Comment:
@kelemena it seems the proxy didn't really solve my issue. Any ideas on this?

Reply:
Since I cannot mock this `new Payment()` call, and if I move this as an input parameter, it will just move the problem from `PaypalProxyTest` to `PaymentManagerTest`.

Reply:
@bearcheese if you follow the same approach as I did, you could have similar solution. 
I suggest to create `Payment` as an output of `ExecutePaymentRequestMapper` and create `PaymentExecution` inside of paypalProxy. With argumentCaptor you could double check apiContext and paymentExecution.

Reply:
Thanks, please see 483f69c